drivers/performance_measurement/app/models/performance_measurement/equity_analysis/builder.rb

```diff
@@ -0,0 +1,209 @@
+module PerformanceMeasurement::EquityAnalysis
+  class Builder
+    include ActiveModel::Model
+
+    validates :metric, presence: true
+    validates :investigate_by, presence: true
+
+    attr_reader :report, :params
+
+    CHART_DATA_KEY_TO_CLASS = {
+      default: PerformanceMeasurement::EquityAnalysis::Data,
+      race: PerformanceMeasurement::EquityAnalysis::RaceData,
+      age: PerformanceMeasurement::EquityAnalysis::AgeData,
+      gender: PerformanceMeasurement::EquityAnalysis::GenderData,
+      household_type: PerformanceMeasurement::EquityAnalysis::HouseholdTypeData,
+    }.freeze
+
+    VEIW_BY_DEFAULT = 'percentage'.freeze
+
+    def initialize(params, report, user)
+      @user = user
+      @report = report
+      @params = params || { view_data_by: VEIW_BY_DEFAULT }
+      @chart_data = chart_data_class.new(self)
+    end
+
+    def chart_data_class
+      CHART_DATA_KEY_TO_CLASS[investigate_by_key]
+    end
+
+    def investigate_by_key
+      key = (investigate_by || '').gsub(/[[:space:]]/, '').underscore.to_sym
+      CHART_DATA_KEY_TO_CLASS.keys.select { |k| k == key }.first || :default
+    end
+
+    def describe_filters
+      rslt = "<span>#{describe_metric} by #{investigate_by}</span></br>"
+      param_rslt = PerformanceMeasurement::EquityAnalysis::Data::INVESTIGATE_BY.keys.map do |key|
+        send("describe_#{key}".to_sym)
+      end
+      param_rslt.push(describe_projects)
+      param_rslt.push(describe_project_type)
+      rslt = "#{rslt} <span>#{param_rslt.reject(&:blank?).join(', ')}</span>"
+      rslt
+    end
+
+    def chart_data
+      {
+        chart_height: @chart_data.chart_height,
+        data: @chart_data.data,
+      }
+    end
+
+    def show_additional_options?
+      metric.present? && investigate_by.present?
+    end
+
+    def metric
+      @params[:metric]&.to_sym
+    end
+
+    def describe_metric
+      metric_options.select { |_, sym| sym == metric }.first.first
+    end
+
+    def investigate_by
+      @params[:investigate_by]
+    end
+
+    def race
+      @params[:race]&.reject(&:blank?) || []
+    end
+
+    def describe_race
+      names = race.map do |key|
+        PerformanceMeasurement::EquityAnalysis::Data::RACES[key]
+      end.reject(&:blank?).join(', ')
+      race.any? ? "Race: #{names}" : ''
```

Comment:
do these need to return '' or can they return nil? If the latter, this could be written as 
```suggestion
      return "Race: #{names}" if race.any? 
```
apply to the other similar lines too